README.md

```diff
@@ -125,7 +136,17 @@ Current state | Next word | Next state
 
 ## A2.4
 
-«replace this with your answer»
+Current state | Next word | Next state
+--------------|-----------|-----------
+    S0        |    the    |     S1
+    S1        |   noun    |     s2
+    s1        |  adjective|     s3
+    s3        | adjective |     s3
+    s3        |   noun    |     s2
+    s2        |   verb    |     s4
+    s4        |   EOI     |     s6
+    s4        |   adverb  |     s5
+    s5        |   EOI     |     s6
 
 
 ## Q2.5 (12 points)
```

Comment:
# 9

There are two issues here. The question asks you to translate the state table into your language, so I was expecting to see it expressed as a data structure, perhaps something like this:

~~~ cpp
#include <iostream>
using namespace std;
#include <map>

typedef int state;
typedef std::map<std::string, state> transitions;
typedef std::map<state, transitions> state_table;

const int END = 99;

state_table states = {
  { 0, {{ "the", 1 }}},
	{ 1, {{ "lazy", 2}, {"smelly", 2}, {"dog", 3}, {"cat", 3}}},
	{ 2, {{ "lazy", 2}, {"smelly", 2}, {"dog", 3}, {"cat", 3}}},
	{ 3, {{ "ate", 4},  {"ran",    4}}},
	{ 4, {{ "slowly", 5}, { "noisily", 5}, {"EOI", END}}},
	{ 5, {{ "EOI", END }}}
};


int getNextState(int currentState, string nextWord) {
	transitions t = states[currentState];
	if (t.count(nextWord))
		return t[nextWord];
	else
		return -1;
}
~~~

Second, the code in interpreter (which is clearly laid out, btw) has a bug. It exits the while loop early when it gets to the EOI. In this case, that also corresponds to the end of the string, but in the general case that's not guaranteed. This is a common error: code that stops after it finds that it has what it is looking for, rather than checking that all it has is what it is looking for.